core/dbt/contracts/graph/unparsed.py

```diff
@@ -830,3 +836,11 @@ class UnparsedUnitTest(dbtClassMixin):
     description: str = ""
     overrides: Optional[UnitTestOverrides] = None
     config: Dict[str, Any] = field(default_factory=dict)
+    versions: Optional[UnitTestNodeVersions] = None
```

Comment:
Is there a reason we couldn't use `dbt_common.helper_types.IncludeExclude` here instead? it would consolidate some of the validation we do below and provides an `.includes` method. Its syntax/error messaging would also be consistent with other features that use include/exclude

Reply:
Didn't even realize that was a thing :). I'll look into using it.

Reply:
The dbt_common IncludeExclude class only handles strings, while the UnitTestNodeVersions has include and exclude for NodeVersion which is a Union of str and float.  Also the dbt_common class handles include and exclude differently (include can be a single string but exclude can't) which wouldn't be necessary for this use case.

It might be possible to use generics to modify the class to use it here, but I'd prefer to investigate that separately. It would be nice to have the same behavior though -- do you think it's worth copying the functionality?

Reply:
Ahhh I see, thanks for digging into it. I think it'd only be worth it to mirror the existing syntax 1:1 but I don't think we'd need to make breaking changes to do that using a generalized IncludeExclude in the future.. so I don't have a strong opinion here